test/extensions/secondorder/diag_ggn/diaggnn_settings.py

```diff
@@ -4,11 +4,238 @@
+        "module_fn": lambda: torch.nn.Sequential(
+            torch.nn.ConvTranspose1d(2, 3, 2, padding=0, dilation=5, stride=3),
+            torch.nn.ReLU(),
+            torch.nn.Flatten(),
+            torch.nn.Linear(72, 5),
+        ),
+        "loss_function_fn": lambda: torch.nn.CrossEntropyLoss(reduction="sum"),
+        "target_fn": lambda: classification_targets((3,), 5),
+    },
+    {
+        "input_fn": lambda: torch.rand(3, 3, 7, 7),
+        "module_fn": lambda: torch.nn.Sequential(
+            torch.nn.ConvTranspose2d(3, 2, 2),
+            torch.nn.ReLU(),
+            torch.nn.Flatten(),
+            torch.nn.Linear(128, 5),
+        ),
+        "loss_function_fn": lambda: torch.nn.CrossEntropyLoss(),
+        "target_fn": lambda: classification_targets((3,), 5),
+    },
+    {
+        "input_fn": lambda: torch.rand(3, 3, 7, 7),
+        "module_fn": lambda: torch.nn.Sequential(
+            torch.nn.ConvTranspose2d(3, 2, 2, bias=False),
+            torch.nn.ReLU(),
+            torch.nn.Flatten(),
+            torch.nn.Linear(128, 5),
+        ),
+        "loss_function_fn": lambda: torch.nn.CrossEntropyLoss(),
+        "target_fn": lambda: classification_targets((3,), 5),
+    },
+    {
+        "input_fn": lambda: torch.rand(3, 2, 9, 9),
+        "module_fn": lambda: torch.nn.Sequential(
+            torch.nn.ConvTranspose2d(2, 4, 2, padding=0, dilation=2, groups=1),
+            torch.nn.ReLU(),
+            torch.nn.Flatten(),
+            torch.nn.Linear(484, 5),
+        ),
+        "loss_function_fn": lambda: torch.nn.CrossEntropyLoss(),
+        "target_fn": lambda: classification_targets((3,), 5),
+    },
+    {
+        "input_fn": lambda: torch.rand(2, 3, 2, 7, 7),
+        "module_fn": lambda: torch.nn.Sequential(
+            torch.nn.ConvTranspose3d(3, 2, 2),
+            torch.nn.ReLU(),
+            torch.nn.Flatten(),
+            torch.nn.Linear(384, 5),
+        ),
+        "loss_function_fn": lambda: torch.nn.CrossEntropyLoss(reduction="mean"),
+        "target_fn": lambda: classification_targets((2,), 5),
+    },
+    {
+        "input_fn": lambda: torch.rand(2, 3, 2, 7, 7),
+        "module_fn": lambda: torch.nn.Sequential(
+            torch.nn.ConvTranspose3d(3, 2, 2, bias=False),
+            torch.nn.ReLU(),
+            torch.nn.Flatten(),
+            torch.nn.Linear(384, 5),
+        ),
+        "loss_function_fn": lambda: torch.nn.CrossEntropyLoss(reduction="mean"),
+        "target_fn": lambda: classification_targets((2,), 5),
+    },
+    {
+        "input_fn": lambda: torch.rand(2, 3, 5, 5, 5),
+        "module_fn": lambda: torch.nn.Sequential(
+            torch.nn.ConvTranspose3d(3, 2, 2, padding=2, dilation=2, stride=2),
+            torch.nn.ReLU(),
+            torch.nn.Flatten(),
+            torch.nn.Linear(686, 5),
+        ),
+        "loss_function_fn": lambda: torch.nn.CrossEntropyLoss(reduction="mean"),
+        "target_fn": lambda: classification_targets((2,), 5),
+    },
+]
```

Comment:
Can you give me an update how these settings differ from the automated simple-cnn tests we have set up?

Reply:
Yes! So each of them differs in the configuration of `torch.nn.ConvTransposed(...)` as in, they have different values of padding, dilation, stride.. etc
I tried to make it automated but that just requires too many additional parameters whereas previous in activations we had only `bias=True/False`